centreon/src/Utility/EnvironmentFileManager.php

```diff
@@ -0,0 +1,196 @@
+                $value = (float) $value;
+            } else {
+                $value = (int) $value;
+            }
+        } elseif (is_string($value)) {
+            if ($value === 'true' || $value === 'false') {
+                $value = filter_var($value, FILTER_VALIDATE_BOOLEAN);
+            } elseif (preg_match('/^-?\d+\.\d+/', $value)) {
+                $value = (float) $value;
+            } else {
+                $value = trim($value);
+            }
+        }
+        $this->variables[trim($key)] = $value;
+    }
+
+    public function clear(): void
+    {
+        $this->variables = [];
+    }
+
+    /**
+     * @param string $key
+     */
+    public function delete(string $key): void
+    {
+        unset($this->variables[$key]);
+    }
+
+    /**
+     * @return array<string, int|float|bool|string>
+     */
+    public function getAll(): array
+    {
+        return $this->variables;
+    }
+
+    /**
+     * @param string $key
+     *
+     * @return int|float|bool|string|null
+     */
+    public function get(string $key): int|float|bool|string|null
+    {
+        return $this->variables[$key] ?? null;
+    }
+
+    /**
+     * @param string|null $environmentFile
+     *
+     * @throws \Exception
+     */
+    public function save(?string $environmentFile = null): void
+    {
+        $this->checkEnvironmentFileDefined($environmentFile);
+        $filePath = $this->environmentFilePath . $this->currentEnvironnementFile;
+        $file = fopen($filePath, 'w');
+        if (! $file) {
+            throw new \Exception(sprintf('Impossible to open file \'%s\'', $filePath));
+        }
+        try {
+            foreach ($this->variables as $key => $value) {
+                if (is_bool($value)) {
+                    $value = $value ? 1 : 0;
+                }
+                fwrite($file, sprintf("%s=%s\n", $key, $value));
+            }
+        } finally {
+            fclose($file);
+        }
+
+        $variables = var_export($this->variables, true);
+        $content = <<<CONTENT
+            <?php
+
+            // This file was generated by Centreon
+
+            return {$variables};
+            CONTENT;
+        file_put_contents($filePath . '.local.php', $content);
+    }
+
+    /**
+     * @param string|null $environmentFile
+     *
+     * @throws \Exception
+     */
+    private function checkEnvironmentFileDefined(?string $environmentFile = null): void
+    {
+        $environmentFile ??= $this->currentEnvironnementFile;
+        if ($environmentFile === null) {
+            throw new \Exception('No environment file defined');
+        }
+    }
+
+    /**
+     * Add the directory separator at the end of the path if it does not exist.
+     *
+     * /dir1/dir2 => /dir1/dir2/
+     *
+     * @param string $path
+     *
+     * @return string
+     */
+    private function addDirectorySeparatorIfNeeded(string $path): string
+    {
+        if (mb_strlen($path) === 0 || $path[-1] !== DIRECTORY_SEPARATOR) {
```

Comment:
Simplier + avoid a useless function call.

```suggestion
        if ($path === '' || $path[-1] !== DIRECTORY_SEPARATOR) {
```